sdk/core/core-client/src/serviceClient.ts

```diff
@@ -426,13 +430,13 @@ export interface ClientPipelineOptions extends InternalPipelineOptions {
  */
 export function createClientPipeline(options: ClientPipelineOptions = {}): Pipeline {
   const pipeline = createPipelineFromOptions(options ?? {});
-
   const credential = options.credentialOptions?.credential;
+  const scopes = options.credentialOptions?.scope || `/.default`;
```

Comment:
Looks like `./scope` on its own isn't valid from my understanding: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#the-default-scope

Reply:
Yeah, maybe we just need to throw in that scenario instead of just sending to /.default?

Reply:
+1 to throwing

Reply:
Or just we don't add the bearerTokenAuthenticationPolicy and log a warning?

Reply:
I think if they pass us a credential without a scope we should throw since the token is useless and the request will fail with some obscure auth error

Reply:
If we don't have a `baseUri` then we can't call the service, so I think we shouldn't end up with `/.default`.  In other words, I think `ServiceClient` should throw when it doesn't have a `baseUri` if it somehow was configured like that

Reply:
I was looking at the service client and it looks like we allow the operationSpec to provide a baseUrl which takes precedence to the serviceClient.baseUrl. SendOperationRequest checks that we have one or the other or throws

![image](https://user-images.githubusercontent.com/21109913/100965516-b8571c00-34df-11eb-8de5-e989c0317b5d.png)

I believe that we are covered in that front with this check in SendOperationRequest, plus the new check that makes sure we end up with a valid credentialScopes.